### PR TITLE
Modal Routing and Sign-Up/Registration Page Styling

### LINKED ISSUE #46: Create Return Component
Most of the modals have a return button for users to go back to the previous page (such as the top left of this [Figma page](https://www.figma.com/file/1GXHLLjElfmieN2GL2olpE/cityfarmslo-high-fidel-v2-(Copy)?node-id=3%3A595)). Create this component and test it, but don't add it to any of the pages yet.

### LINKED ISSUE #48: Implement Modal View
Using the work from the [design modal PR](https://github.com/hack4impact-calpoly/city-farm-slo/pull/38), change the signup and confirmation pages so they look like the [Figma mockups](https://www.figma.com/file/Wf8y2Rb6WzgAnbEb8eWAjk/cityfarmslo-high-fidel-v3) with the background image, rounded corners, and blue backgrounds. 

Please reach out if you have any questions!

### LINKED ISSUE #66: Add back button to site
Add the back button from PR #56 to the bottom left page of each page that has it on the [Figma](https://www.figma.com/file/Wf8y2Rb6WzgAnbEb8eWAjk/cityfarmslo-high-fidel-v3). Also please change the RegistrationComplete component so the return to City Farm SLO button links back [here](https://www.cityfarmslo.org/volunteer)

### LINKED ISSUE #82: The Event Card on the SignUp page is displaying to the right of the title instead of below

### PR DESCRIPTION
**Pull request by ryanhu021**
Placed sign-up and registration complete inside model, added routing to buttons.

### CODE REVIEW

File: frontend/src/components/HomeModal.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import Modal from "@mui/material/Modal";
+// import { isMobile } from "react-device-detect";
+
+export default function HomeModal({ children, open, handleClose }) {
+  const ModalContainer = styled.div`
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    min-width: 50%;
```
Comment (vwinstea) [inline frontend/src/components/HomeModal.jsx]: Could you make the modal take up more of the page? Maybe 75%?

File: frontend/src/components/HomePage/home.jsx [hunk 2]
```diff
@@ -190,6 +204,22 @@ export default function Home({ selectedEvent, setEvent }) {
...
+            />
+          </Route>
+          <Route path="/registration-complete">
+            <RegistrationComplete
+              selectedEvent={selectedEvent}
+              handleModalClose={handleModalClose}
+            />
+          </Route>
+        </Switch>
+      </HomeModal>
```
Comment (vwinstea) [inline frontend/src/components/HomePage/home.jsx]: Nice work figuring this out!

File: frontend/src/components/UserSignUp/SignUpForm.jsx [hunk 3]
```diff
@@ -120,43 +158,86 @@ function SignUpForm({ selectedEvent }) {
...
     console.log(values);
     reset();
+    history.push("/registration-complete");
   };
 
   // info for required entries
   const rEntries = [
     { name: "name", label: "Name" },
     { name: "email", label: "Email" },
-    { name: "number", label: "Phone Number" },
```
Comment (vwinstea) [inline frontend/src/components/UserSignUp/SignUpForm.jsx:129]: Why isn't phone required anymore?

**Review by vwinstea — CHANGES_REQUESTED** [review]
Excellent work! This is looking really good and I can't wait to show the team next week. I only have one request: could you make the modal a bit bigger (75% of page)? There's some changes we're planning on making to the sign up page and right now they won't fit with the current size. Thanks for working on this task for a couple weeks, it was worth the wait!

File: frontend/src/components/UserSignUp/SignUpForm.jsx (hunk 3 above)
Comment (ryanhu021) [reply frontend/src/components/UserSignUp/SignUpForm.jsx:129]: The array was used to render the fields in a single section, but I had to separate the phone and notes parts into separate sections.

**Review by vwinstea — APPROVED** [review]
Merge conflicts resolved!

**Pull request merged**